source/extensions/omni.isaac.lab/omni/isaac/lab/envs/manager_based_env.py

```diff
@@ -245,20 +257,78 @@ def reset(self, seed: int | None = None, options: dict[str, Any] | None = None)
         Returns:
             A tuple containing the observations and extras.
         """
+        if env_ids is None:
+            env_ids = torch.arange(self.num_envs, dtype=torch.int64, device=self.device)
+
+        # trigger recorder terms for pre-reset calls
+        self.recorder_manager.record_pre_reset(env_ids)
+
         # set the seed
         if seed is not None:
             self.seed(seed)
 
         # reset state of scene
-        indices = torch.arange(self.num_envs, dtype=torch.int64, device=self.device)
-        self._reset_idx(indices)
+        self._reset_idx(env_ids)
+        self.scene.write_data_to_sim()
```

Comment:
by calling this, do you still need the `update_articulation_kinematics()` change from https://github.com/isaac-sim/IsaacLab/pull/1276?

Reply:
Yes, `write_data_to_sim()` doesn't seem to make the sensor data update.